tests/all/wasi_testsuite.rs

```diff
@@ -122,8 +118,8 @@ fn build_command<P: AsRef<Path>>(module: P, extra_flags: &[&str], spec: &Spec) -
     cmd.args(extra_flags);
     if let Some(dirs) = &spec.dirs {
         for dir in dirs {
-            cmd.arg("--dir");
-            cmd.arg(parent_dir.join(dir));
+            cmd.arg("--mapdir");
+            cmd.arg(format!("{}::{}", dir, parent_dir.join(dir).display()));
```

Comment:
Instead of passing the raw directory in the spec test suite to each test, could the directory perhaps be copied to a temporary location and have the test run against that? That way the temporary directory clean up would removing any extra files and the `clean_garbage` filtering may not be necessary?

Alternatively though explicitly cleaning out files I think is ok, and if this is how the test suite runs upstream I think it's fine to mirror that as well.

Reply:
Well, right now I believe the other `wasi-testsuite` test runner (the Python one) expects to clean up any `*.cleanup` files so I propose we stick with that for now. I opened https://github.com/WebAssembly/wasi-testsuite/pull/74 to actually right down what a test runner should do and I added a comment there about this. How about we change this logic once we decide what to do there?

Reply:
Sounds good to me 👍